phoenix-core/src/it/java/org/apache/phoenix/end2end/UpgradeIT.java

```diff
@@ -697,21 +698,33 @@ public void testLastDDLTimestampBootstrap() throws Exception {
             conn.createStatement().execute(
                 "CREATE VIEW " + fullViewName + " AS SELECT * FROM " + fullTableName);
 
+            conn.createStatement().execute("CREATE INDEX " + indexName + " ON " + fullTableName + " (KV1) ASYNC");
+
             //Now we null out any existing last ddl timestamps
             nullDDLTimestamps(conn);
 
             //now get the row timestamps for each header row
             long tableTS = getRowTimestampForMetadata(conn, schemaName, tableName,
                 PTableType.TABLE);
             long viewTS = getRowTimestampForMetadata(conn, schemaName, viewName, PTableType.VIEW);
+            long indexTS = getRowTimestampForMetadata(conn, schemaName, indexName, PTableType.INDEX);
 
-            UpgradeUtil.bootstrapLastDDLTimestamp(conn.unwrap(PhoenixConnection.class));
+            // bootstrap last ddl timestamp for tables and views
+            UpgradeUtil.bootstrapLastDDLTimestampForTablesAndViews(conn.unwrap(PhoenixConnection.class));
             long actualTableTS = getLastTimestampForMetadata(conn, schemaName, tableName,
                 PTableType.TABLE);
             long actualViewTS = getLastTimestampForMetadata(conn, schemaName, viewName,
                 PTableType.VIEW);
+            long actualIndexTS = getLastTimestampForMetadata(conn, schemaName, indexName,
+                    PTableType.INDEX);
             assertEquals(tableTS, actualTableTS);
             assertEquals(viewTS, actualViewTS);
+            assertEquals(0L, actualIndexTS);
```

Comment:
should actualIndexTS be NULL or 0? I think it should be null.

Reply:
It is returned using a `rs.getLong()` call which returns 0 if the actual value was sql null.